lib/config.js

```diff
@@ -1653,17 +1673,25 @@ util.runStrictnessChecks = function (config) {
     _warnOrThrow("NODE_ENV value of '"+NODE_ENV+"' is ambiguous.");
   }
 
-  function _warnOrThrow (msg) {
-    var beStrict = process.env.NODE_CONFIG_STRICT_MODE;
-    var prefix = beStrict ? 'FATAL: ' : 'WARNING: ';
-    var seeURL = 'See https://github.com/lorenwest/node-config/wiki/Strict-Mode';
+  function _warnOrThrow (msg, neverThrow) {
+    var prefix = beStrict && !neverThrow ? 'FATAL: ' : 'WARNING: ';
 
     console.error(prefix+msg);
-    console.error(prefix+seeURL);
 
-    // Accept 1 and true as truthy values. When set via process.env, Node.js casts them to strings.
-    if (["true", "1"].indexOf(beStrict) >= 0) {
-      throw new Error(prefix+msg+' '+seeURL);
+    // Unless this is a warning that's never fatal, let the user know about strict mode.
+    if (!neverThrow) {
+      var modeInfo = beStrict ? 'Strict mode is on, aborting.' : 'Strict mode is off, continuing.';
+      var modeMessage =  modeInfo + ' See https://github.com/lorenwest/node-config/wiki/Strict-Mode';
+      console.error(prefix+modeMessage);
+      if (beStrict) {
+        throw new Error(prefix+msg+' '+modeMessage);
+      }
+    }
+
+    // If we're not in strict mode or this is a message that's safe to disable, let the user
+    // know that they can turn off warnings altogether
+    if (!beStrict || neverThrow) {
+      console.error(prefix+'To disable this warning, set SUPPRESS_NO_CONFIG_WARNING in the environment.');
```

Comment:
> To clarify, we do want the variable to suppress the specific warnings, even if a default config is present, for the cases of NODE_ENV and NODE_APP_INSTANCE settings that don't match config files, correct?

Yes, if strict mode is not on. If strict mode is on, it's a fatal. error. This was points 1 and 2 of my "spec" about this should work:

1. If SUPPRESS_NO_CONFIG_WARNING is set and NODE_ENV does not match a config file, the warning will suppressed when strict mode is off, but still be a fatal error when strict mode is on.
2. If SUPPRESS_NO_CONFIG_WARNING is set and NODE_APP_INSTANCE does not match a config file, the warning will suppressed when strict mode is off, but still be a fatal error when strict mode is on.
3. If SUPPRESS_NO_CONFIG_WARNING is set, there are no config files, and NODE_ENV is set to "default" or "local", a warning will be suppressed when strict mode is off, but still be a fatal error when strict mode is on.

> we want to migrate to a single flag that suppresses all warnings during the initialization process

I don't want that. All of the warnings are there for a reason. I like the idea that you to explicitly suppress each class. With a single giant "disable all" flag, you might end up disabling a warning you've never seen before and didn't intend to disable-- some issue that would bite you only after you set the global "disable all warnings" flag. 

If anything, we should prioritize in a different direction: making it easy for your configuration to *quit generating warnings because the configuration is safe*.